src/core/server/saved_objects/service/lib/aggregations/validation_utils.ts

```diff
@@ -24,15 +24,16 @@ export const isRootLevelAttribute = (
   allowedTypes: string[]
 ): boolean => {
   const splits = attributePath.split('.');
-  if (splits.length !== 2) {
+  if (splits.length <= 1) {
     return false;
   }
 
-  const [type, fieldName] = splits;
-  if (allowedTypes.includes(fieldName)) {
+  const [type, ...fieldName] = splits;
+  if (allowedTypes.includes(fieldName.join('.'))) {
     return false;
   }
-  return allowedTypes.includes(type) && fieldDefined(indexMapping, fieldName);
+
+  return allowedTypes.includes(type) && fieldDefined(indexMapping, fieldName.join('.'));
```

Comment:
Can you add a test asserting that you can't access a property via `{type}.{type}.{attr}`, e.g that accessing `dashboard.dashboard.title` or `alert.alert.actions.group` is not allowed?